### PR TITLE
V1.0.0

### PR DESCRIPTION
**Pull request by SC426**
### Forked from Conda Forge
[Upstream](https://github.com/pypa/pyproject-hooks/tree/v1.0.0)
[Changelog](https://pyproject-hooks.readthedocs.io/en/latest/changelog.html)

### Actions
- Remove `noarch`
- Add `--no-deps` to build script
- Unpin dependencies in host section
- Add `setuptools` and `wheel`
- Linter fixes

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,17 +10,19 @@ source:
...
+    - python 
     - pip
-    - flit-core >=3.2, <4
+    - setuptools
+    - wheel
+    - flit-core
   run:
-    - python >=3.7
+    - python
     - tomli >=1.1.0
```
Comment (ELundby45) [inline recipe/meta.yaml]:
```suggestion
    - tomli >=1.1.0.  # [py<311]
```

https://github.com/pypa/pyproject-hooks/blob/v1.0.0/pyproject.toml#L13

Comment (SC426) [reply recipe/meta.yaml]: Fixed!

**Review by ELundby45 — APPROVED** [review]
As long as my last suggestion doesn't break the builds. LGTM 😄

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,18 +10,20 @@ source:
...
 
 requirements:
   host:
-    - python >=3.7
+    - python 
     - pip
-    - flit-core >=3.2, <4
+    - setuptools
+    - wheel
+    - flit-core
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:23]: so it does not have pinning because its python dependency, right ?

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,18 +10,20 @@ source:
   sha256: f271b298b97f5955d53fb12b72c1fb1948c22c1a6b70b315c54cedaca0264ef5
 
 build:
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [py<38]
```
Comment (akabanovs) [inline recipe/meta.yaml]: Upstream supports python 3.7 and above. 
https://github.com/pypa/pyproject-hooks/blob/v1.0.0/pyproject.toml#L11

File: recipe/meta.yaml (hunk 2 above)
Comment (SC426) [reply recipe/meta.yaml:23]: I think we actually want to have a more exact pinning in the host section (according to the linter)

File: recipe/meta.yaml (hunk 3 above)
Comment (SC426) [reply recipe/meta.yaml]: Fixed!

**Pull request merged**